warehouse/internal/model/upload.go

```diff
@@ -21,45 +24,86 @@ const (
 	Aborted                   = "aborted"
 )
 
-type UploadJob struct {
-	Warehouse    warehouseutils.Warehouse
-	Upload       Upload
-	StagingFiles []*StagingFile
-}
+var ErrUploadNotFound = errors.New("upload not found")
 
 type Upload struct {
-	ID                   int64
-	Namespace            string
-	WorkspaceID          string
-	SourceID             string
-	SourceType           string
-	SourceCategory       string
-	DestinationID        string
-	DestinationType      string
-	StartStagingFileID   int64
-	EndStagingFileID     int64
-	StartLoadFileID      int64
-	EndLoadFileID        int64
-	Status               string
-	UploadSchema         warehouseutils.SchemaT
-	MergedSchema         warehouseutils.SchemaT
-	Error                json.RawMessage
-	Timings              []map[string]string
-	FirstAttemptAt       time.Time
-	LastAttemptAt        time.Time
-	Attempts             int64
-	Metadata             json.RawMessage
-	FirstEventAt         time.Time
-	LastEventAt          time.Time
-	UseRudderStorage     bool
+	ID          int64
+	WorkspaceID string
+
+	Namespace       string
+	SourceID        string
+	DestinationID   string
+	DestinationType string
+	Status          string
+	Error           json.RawMessage
+	FirstEventAt    time.Time
+	LastEventAt     time.Time
+
+	UseRudderStorage bool
+	SourceBatchID    string
+	SourceTaskID     string
+	SourceTaskRunID  string
+	SourceJobID      string
+	SourceJobRunID   string
+	LoadFileType     string
+	NextRetryTime    time.Time
+	Priority         int
+	Retried          bool
+
+	StagingFileStartID int64
+	StagingFileEndID   int64
+
+	LoadFileStartID int64
+	LoadFileEndID   int64
+
+	Timings        Timings
+	FirstAttemptAt time.Time
+	LastAttemptAt  time.Time
+	Attempts       int64
+
+	UploadSchema Schema
+	MergedSchema Schema
+}
+
+type Timings []map[string]time.Time
+
+type Schema = warehouseutils.SchemaT
```

Comment:
Curious to know, since we already have a type schema defined in utils, We defined one more schema type here as well pointing to that one.

Reply:
I did this `Schema = warehouseutils.SchemaT ` to avoid having to modify all the assurances of SchemaT. Also, since the util package is using Schema as well it would result in a circular dependency, thus more changes would be required.